core/chains/evm/txmgr/client.go

```diff
@@ -38,12 +39,12 @@ func (c *evmTxmClient) ConfiguredChainID() *big.Int {
 	return c.client.ConfiguredChainID()
 }
 
-func (c *evmTxmClient) BatchSendTransactions(ctx context.Context, txStore TxStore, attempts []EvmTxAttempt, batchSize int, lggr logger.Logger) (codes []clienttypes.SendTxReturnCode, txErrs []error, err error) {
+func (c *evmTxmClient) BatchSendTransactions(ctx context.Context, updateBroadcastTime func(now time.Time, txIDs []int64) error, attempts []EvmTxAttempt, batchSize int, lggr logger.Logger) (codes []clienttypes.SendTxReturnCode, txErrs []error, err error) {
```

Comment:
This seems badly designed, especially now that we've extracted things in to generic components.
I mean, the call which actually does a batchSendTx(), in the chain specific code, shouldn't be responsible for updating persistent state. Infact it shouldn't even be aware of it.

Ideally, the caller in the generic TXM side, should note all successful broadcasted txs, and then update the state and DB there itself.

Does that make sense? Maybe for a follow-up PR?

Reply:
agreed, the client shouldn't care about the state persistence - i'll make an additional clean up ticket